In [2]:
import pandas as pd
titanic_df=pd.read_csv("F:\github\Data preprocessing\Titanic_data.csv",delimiter=",", header=0)


<>:2: SyntaxWarning: "\g" is an invalid escape sequence. Such sequences will not work in the future. Did you mean "\\g"? A raw string is also an option.
<>:2: SyntaxWarning: "\g" is an invalid escape sequence. Such sequences will not work in the future. Did you mean "\\g"? A raw string is also an option.
C:\Users\User\AppData\Local\Temp\ipykernel_24196\153412453.py:2: SyntaxWarning: "\g" is an invalid escape sequence. Such sequences will not work in the future. Did you mean "\\g"? A raw string is also an option.
  titanic_df=pd.read_csv("F:\github\Data preprocessing\Titanic_data.csv",delimiter=",", header=0)


In [3]:
titanic_df.head()

,PassengerId,Survived,Pclass,Name,Sex,Age,SibSp,Parch,Ticket,Fare,Cabin,Embarked
0,1,0,3,"Braund, Mr. Owen Harris",male,22.0,1,0,A/5 21171,7.2500,NaN,S
1,2,1,1,"Cumings, Mrs. John Bradley (Florence Briggs Th...",female,38.0,1,0,PC 17599,71.2833,C85,C
2,3,1,3,"Heikkinen, Miss. Laina",female,26.0,0,0,STON/O2. 3101282,7.9250,NaN,S
3,4,1,1,"Futrelle, Mrs. Jacques Heath (Lily May Peel)",female,35.0,1,0,113803,53.1000,C123,S
4,5,0,3,"Allen, Mr. William Henry",male,35.0,0,0,373450,8.0500,NaN,S


In [4]:
titanic_df.shape

(891, 12)

In [5]:
titanic_df.info()

<class 'pandas.DataFrame'>
RangeIndex: 891 entries, 0 to 890
Data columns (total 12 columns):
 #   Column       Non-Null Count  Dtype  
---  ------       --------------  -----  
 0   PassengerId  891 non-null    int64  
 1   Survived     891 non-null    int64  
 2   Pclass       891 non-null    int64  
 3   Name         891 non-null    str    
 4   Sex          891 non-null    str    
 5   Age          714 non-null    float64
 6   SibSp        891 non-null    int64  
 7   Parch        891 non-null    int64  
 8   Ticket       891 non-null    str    
 9   Fare         891 non-null    float64
 10  Cabin        204 non-null    str    
 11  Embarked     889 non-null    str    
dtypes: float64(2), int64(5), str(5)
memory usage: 83.7 KB


In [6]:
#specific INformation
#stastical info mean median{Q2} mode std min max
titanic_df[["Age","Fare"]].mean()

Age     29.699118
Fare    32.204208
dtype: float64

In [7]:
titanic_df[["Age","Fare"]].median()

Age     28.0000
Fare    14.4542
dtype: float64

In [8]:
titanic_df[["Age","Fare"]].describe()

,Age,Fare
count,714.000000,891.000000
mean,29.699118,32.204208
std,14.526497,49.693429
min,0.420000,0.000000
25%,20.125000,7.910400
50%,28.000000,14.454200
75%,38.000000,31.000000
max,80.000000,512.329200


In [9]:
titanic_df.Embarked.mode()
#write description 

0    S
Name: Embarked, dtype: str


#DEscription
Analysis of Age and Fare
Age has 714 valid values out of 891 passengers, meaning 177 passengers have missing age data.
The average age is approximately 29.7 years, indicating that the typical passenger was around 30 years old.
The median age is 28 years, which is slightly lower than the mean, suggesting a small influence from older passengers.
The youngest passenger was only 0.42 years old (about 5 months), while the oldest passenger was 80 years old.
The standard deviation of Age is 14.53, showing considerable variation in passengers' ages.
For Fare, all 891 passengers have valid values, so there are no missing Fare values.
The average fare was approximately 32.20, while the median fare was only 14.45.
Since the mean fare (32.20) is much higher than the median fare (14.45), the Fare data is right-skewed, meaning a small number of passengers paid very high fares.
The lowest fare was 0, while the highest fare was 512.33, showing a very large difference between the cheapest and most expensive tickets.
The middle 50% of passengers paid between approximately 7.91 and 31.00.
Overall, Age is relatively spread out, while Fare has strong variation and outliers, particularly among passengers who paid very expensive tickets.

In [10]:
#counts
titanic_df.value_counts("Survived")

Survived
0    549
1    342
Name: count, dtype: int64

In [11]:
titanic_df.Survived

0      0
1      1
2      1
3      1
4      0
      ..
886    0
887    1
888    0
889    1
890    0
Name: Survived, Length: 891, dtype: int64

In [12]:
#counts
titanic_df.value_counts("Embarked", normalize=True, ascending=True)

Embarked
Q    0.086614
C    0.188976
S    0.724409
Name: proportion, dtype: float64

In [14]:
#counts
titanic_df.value_counts("Embarked", normalize=False, dropna=False)

Embarked
S      644
C      168
Q       77
NaN      2
Name: count, dtype: int64

Interestingly, Large group of passangers embarked from Southampton

In [16]:
titanic_df.value_counts(["Sex","Survived"], normalize=True)

Sex     Survived
male    0           0.525253
female  1           0.261504
male    1           0.122334
female  0           0.090909
Name: proportion, dtype: float64

In [18]:
titanic_df[titanic_df.Sex=="female"].value_counts("Survived")

Survived
1    233
0     81
Name: count, dtype: int64

In [ ]:
#Embarked from S, Fare DIstribution?
filtered=titanic_df[titanic_df.Embarked=="S"]["Fare"].describe()
filtered



count    644.000000
mean      27.079812
std       35.887993
min        0.000000
25%        8.050000
50%       13.000000
75%       27.900000
max      263.000000
Name: Fare, dtype: float64

In [32]:
#Embarked from S and gender Male, survival rate?
filtered = titanic_df[
    (titanic_df.Embarked == "S") & 
    (titanic_df["Sex"] == "male")
]

filtered_survived = filtered["Survived"].mean()
filtered_survived

np.float64(0.1746031746031746)